src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,325 @@
+        _;
+    }
+
+    constructor(
+        ISlasher _slasher,
+        IServiceManager _serviceManager,
+        IStakeRegistry _stakeRegistry,
+        IBLSPubkeyRegistry _blsPubkeyRegistry,
+        IIndexRegistry _indexRegistry
+    ) {
+        slasher = _slasher;
+        serviceManager = _serviceManager;
+        stakeRegistry = _stakeRegistry;
+        blsPubkeyRegistry = _blsPubkeyRegistry;
+        indexRegistry = _indexRegistry;
+    }
+
+    function initialize(OperatorSetParam[] memory _operatorSetParams) external initializer {
+        // the stake registry is this contract itself
+        registries.push(address(stakeRegistry));
+        registries.push(address(blsPubkeyRegistry));
+        registries.push(address(indexRegistry));
+
+        // set the operator set params
+        require(IVoteWeigher(address(stakeRegistry)).quorumCount() == _operatorSetParams.length, "BLSIndexRegistryCoordinator: operator set params length mismatch");
+        for (uint8 i = 0; i < _operatorSetParams.length; i++) {
+            quorumOperatorSetParams[i] = _operatorSetParams[i];
+        }
+    }
+
+    /// @notice Returns task number from when `operator` has been registered.
+    function getFromTaskNumberForOperator(address operator) external view returns (uint32) {
+        return _operators[operator].fromTaskNumber;
+    }
+
+    /// @notice Returns the operator struct for the given `operator`
+    function getOperator(address operator) external view returns (Operator memory) {
+        return _operators[operator];
+    }
+
+    /// @notice Returns the operatorId for the given `operator`
+    function getOperatorId(address operator) external view returns (bytes32) {
+        return _operators[operator].operatorId;
+    }
+
+    /// @notice Returns the quorum bitmap for the given `operatorId` at the given `blockNumber` via the `index`
+    function getQuorumBitmapByOperatorIdAtBlockNumberByIndex(bytes32 operatorId, uint32 blockNumber, uint256 index) external view returns (uint192) {
+        QuorumBitmapUpdate memory quorumBitmapUpdate = _operatorIdToQuorumBitmapHistory[operatorId][index];
+        require(
+            quorumBitmapUpdate.updateBlockNumber <= blockNumber, 
+            "BLSRegistryCoordinator.getQuorumBitmapByOperatorIdAtBlockNumberByIndex: quorumBitmapUpdate is from after blockNumber"
+        );
+        require(
+            quorumBitmapUpdate.nextUpdateBlockNumber > blockNumber, 
+            "BLSRegistryCoordinator.getQuorumBitmapByOperatorIdAtBlockNumberByIndex: quorumBitmapUpdate is from before blockNumber"
+        );
+        return quorumBitmapUpdate.quorumBitmap;
+    }
+
+    /// @notice Returns the current quorum bitmap for the given `operatorId`
+    function getCurrentQuorumBitmapByOperatorId(bytes32 operatorId) external view returns (uint192) {
+        return _operatorIdToQuorumBitmapHistory[operatorId][_operatorIdToQuorumBitmapHistory[operatorId].length - 1].quorumBitmap;
```

Comment:
any issue with this reverting for length 0 arrays? I'm assuming it's OK but wanted to note the possibility

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/88d0c50a4d29ace622cdf2043699c68caffd5f74